src-electron/rest/endpoint.js

```diff
@@ -133,6 +133,29 @@ function httpPatchEndpoint(db) {
   return async (request, response) => {
     let context = request.body
     let sessionIdexport = request.zapSessionId
+    let parentEndpointIdentifier = context.parentEndpointIdentifier
+    let parentEndpointRef = await queryEndpoint.getParentEndpointRef(
+      db,
+      parentEndpointIdentifier,
+      sessionIdexport
+    )
+    if (parentEndpointRef == null && parentEndpointIdentifier != null) {
+      parentEndpointIdentifier = null
+      notification.setNotification(
```

Comment:
You should immediately return the 404 or some other HTTP error here. This is not what we should use notifications for!

Notifications are for asynchronous issues. For immediate failures you should immediatelly send back the appropriate HTTP error code.

Reply:
good idea, will do, thanks

Reply:
@tecimovic I changed it so it errors if the user chooses a Parent that does not exist.

That said, I left in the Notification so the user gets a pop up message stating why...